middlewares/datadog.go

```diff
@@ -11,7 +11,8 @@ import (
 	"github.com/go-kit/kit/metrics/dogstatsd"
 )
 
-var _ Metrics = (Metrics)(nil)
+var _ Metrics = (*Datadog)(nil)
+var _ RetryMetrics = (*Datadog)(nil)
```

Comment:
Opps... must've been a glitch while rebasing between various branches...   should be part of #1884